ShoppingCart.java

```diff
@@ -0,0 +1,133 @@
+package ShoppingCart;
+
+import java.util.*;
+
+public class ShoppingCart {
+    private Map<Item, Integer> itemList;
+    private int currentTotalPrice;
+    private BuyLaterList buyLater;
+    private User user;
+
+    public ShoppingCart(User u) {
+        user = u;
+        itemList = new HashMap<Item, Integer>();
+        currentTotalPrice = 0;
+        buyLater = new BuyLaterList();
+    }
+
+    public void addItem(Item item) {
+        if (itemList.containsKey(item)) {
+            itemList.put(item, itemList.get(item) + 1);
+        } else {
+            itemList.put(item, 1);
+        }
+        currentTotalPrice = currentTotalPrice + item.getPrice();
+    }
+
+    public void deleteItem(Item item) throws Exception {
+        if (!itemList.containsKey(item)) {
+            throw new Exception("Item does not exit!");
+        }
+        currentTotalPrice = currentTotalPrice - item.getPrice() * itemList.get(item);
+        itemList.remove(item);
+    };
+
+    public void itemAmountPlusOne(Item item) throws Exception{
+        if (!itemList.containsKey(item)) {
+            addItem(item);
+        }
+        if (itemList.get(item) == item.getStockAmount()) {
```

Comment:
このほうが安全かも（2個以上追加されるメソッドが生まれた時対策）
```suggestion
        if (itemList.get(item) >= item.getStockAmount()) {
```